core/execution/market.go

```diff
@@ -3166,16 +3207,34 @@ func (m *Market) commandLiquidityAuction(ctx context.Context) {
 func (m *Market) tradingTerminated(ctx context.Context, tt bool) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.mkt.State = types.MarketStateTradingTerminated
-	m.mkt.TradingMode = types.MarketTradingModeNoTrading
-	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
-	m.stateChanged = true
 
-	if m.settlementPriceInMarket == nil {
-		m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+	if m.mkt.State != types.MarketStateProposed && m.mkt.State != types.MarketStatePending {
+		m.mkt.State = types.MarketStateTradingTerminated
+		m.mkt.TradingMode = types.MarketTradingModeNoTrading
+		m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
+		m.stateChanged = true
+
+		if m.settlementPriceInMarket == nil {
+			m.log.Debug("no settlement price", logging.MarketID(m.GetID()))
+			return
+		}
+		m.settlementPriceWithLock(ctx)
+	} else {
+		for party := range m.parties {
+			_, err := m.CancelAllOrders(ctx, party)
+			if err != nil {
+				m.log.Debug("could not cancel orders for party", logging.PartyID(party), logging.Error(err))
+			}
+			err = m.closeCancelledMarket(ctx, m.timeService.GetTimeNow())
```

Comment:
this is definitely not right. why is this done for every party? separately why is this done at all? if the only case when a market gets into cancelled state is when it is being terminated before it is being enacted, then there can't be any orders - so there's still some cleanup to do but I don't think you need everything in closeCancelledMarket

Reply:
There would be orders eventually. The market is in opening auction between during the PENDING state (which is meant to end when the market is enacted). This order would require margin etc.

Reply:
the way the conditions are written now, yes, but I'm asking if this is what we want - i.e. why terminating during opening auction is treated similarly to terminating before enactment